configuration/developing-community-health-applications.md

```diff
@@ -455,28 +472,30 @@ More complex targets can be written using the full set of properties for targets
 | `type` | `'count'` or `'percent'` | The type of the widget. | yes |
 | `goal` | `integer` | For targets with `type: 'percent'`, an integer from 0 to 100. For `type: 'count'`, any positive number. If there is no goal, put -1. | yes |
 | `appliesTo` | `'contacts'` or `'reports'` | Do you want to count reports or contacts? This attribute controls the behavior of other attributes herein. | yes |
 | `appliesToType` | If `appliesTo: 'reports'`, an array of form codes. If `appliesTo: 'contacts'`, an array of contact types. | Filters the contacts or reports for which `appliesIf` will be evaluated. For example, `['person']` or `['clinic', 'health_center']`. For example, `['pregnancy']` or `['P', 'pregnancy']`. | no |
 | `appliesIf` | `function(contact, report)` | If `appliesTo: 'contacts'`, this function is invoked once per contact and `report` is undefined. If `appliesTo: 'reports'`, this function is invoked once per report. Return true to count this document. For `type: 'percent'`, this controls the denominator. | no |
-| `passesIf` | `function(contact, report)` | For `type: 'percent'`, return true to increment the numerator. | yes, if `type: 'percent'` |
-| `date` | `'reported'` or `'now'` | When `'reported'`, the target will include documents with a `reported_date` within the current month. When `'now'`, target includes all documents. Default is `'reported'`. | no |
-| `idType` | `'report'` or `'contact'` or `function(contact, report)` | The target's values are incremented once per unique ID. To count individual contacts that meet the criteria, use `'contact'`. To count multiple reports per contact, use `'report'`. If neither keyword suits your needs you can write your own function to provide the ID. | no |
-
-### Additional code
-Helper variables and functions can be defined in `nools-extras.js` to keep the target definitions easy to read and manage. To enable reuse of common code, `nools-extras.js` file is shared by both the Tasks and Targets.
+| `passesIf` | `function(contact, report)` | For `type: 'percent'`, return true to increment the numerator. | yes, if `type: 'percent'`. forbidden when `groupBy` is defined |
+| `date` | `'reported'` or `'now'` or `function(contact, report)` | When `'reported'`, the target will count documents with a `reported_date` within the current month. When `'now'`, target includes all documents. A function can be used to indicate when the document should be counted. Default is `'reported'`. | no |
+| `idType` | `'report'` or `'contact'` or `function(contact, report)` | The target's values are incremented once per unique ID. To count individual contacts that meet the criteria, use `'contact'`. To count multiple reports per contact, use `'report'`. If neither keyword suits your needs you can write your own function to provide the ID(s). | no |
```

Comment:
This is an anchor of the example given below, so may need a clearer explanation when one of the string types is not used. Would this be accurate?

```suggestion
| `idType` | `'report'` or `'contact'` or `function(contact, report)` | The target's values are incremented once per unique ID. To count individual contacts that meet the criteria, use `'contact'`. Use `'report'` to count all reports, even if there are multiple for a single contact. If you need more than a single count for each contact or report, a custom function can be used returning an array with unique IDs — for each element to count. | no |
```

Reply:
What do you think of rename `idType` to something like `sample`.  If we have `sample: 'reports'` then every report causes the target to increment. If we have `sample: 'contact'` then multiple reports for a contact with value true for `appliesIf` will cause the target to increment once. Then perhaps a deeper section is required to explain custom samples (function). 

If I were to write a deeper section about custom samples, the details authors need to know is: 
* A target's displayed score is the aggregation of its "samples"
* Each sample has an ID and each unique ID causes the target to increment by one.
* Custom samples are useful in two scenarios: 1) incrementing the target multiple times based on a single document, and 2) avoiding redundant counts by causing IDs to conflict. 

**incrementing the target multiple times based on a single document**
* You can return an array of strings to emit many samples
* Maybe a quick example of incrementing a target many times based on data in a single doc. Like a stock scenario like those from Togo?

**avoiding redundant counts by causing IDs to conflict**
When multiple samples have the same ID:
* If a single contact causes multiple samples with the same ID, the one from the oldest report is scored. For example, when `sample: 'contact'` multiple reports may yield samples but they all share the same ID (the id of the contact's document) so the target displays the number of contacts with samples.
* If two different contacts cause multiple samples with the same ID, the older contact's sample is scored. For example, using the contact's parent will count the number of places for which a child contact has some condition happen. Or using the date of a reported_date timestamp as the sample ID will count the number of days any contact has any report with some condition.

What do you think of this outline? In general, what do you think of the naming to use "sample"? Would you add/remove any details? If you like it, I can draft something up (probaby a new PR?).